File: internal/tools/format.go
package tools

import (
	"fmt"
	"strings"
)

func truncateLines(output string, maxLines int, label string) string {
	lines := strings.Split(output, "\n")
	if len(lines) <= maxLines {
		return output
	}
	return strings.Join(lines[:maxLines], "\n") + "\n" + muted(fmt.Sprintf("... (%d %s)", len(lines)-maxLines, label))
}

File: kai/tools/format.py
from kai.tools.colors import muted


def truncate_lines(output: str, max_lines: int, label: str) -> str:
    lines = output.split("\n")
    if len(lines) <= max_lines:
        return output
    return (
        "\n".join(lines[:max_lines])
        + "\n"
        + muted(f"... ({len(lines) - max_lines} {label})")
    )
